### PR TITLE
Remove Pods From Demo App and Added JustTweak Dependency as swift Package

### PR DESCRIPTION
**Pull request by mariumhassan**
Remove Pods From Demo App and Added JustTweak Dependency as swift Package

## Description
Previously the Example App was using JustTweak depency through Pods, remove CocoaPods from the whole project and added this dependency through swift package.
Renamed the Example App as "Demo App".
Removed the Unit tests from "Demo App" and moved them under JustTweak Package.
Added "TweakAccessorGenerator" as a an executable Target in main package ("JustTweak").
Made Changes in GitHub Actions, to run tests from package instead of directly from the workspace.

## Motivation and Context
Did this change to remove CocoaPods and move to usage of SPM.
Organise the code into a more structured form.

## How Has This Been Tested?
Build and Run the project.
Run all tests through fastlane and also compiled and run in Xcode.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .build/plugins/cache/Generate_Manual-state.json [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    "5.6.0",
+    "-I",
+    "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/pm/PluginAPI",
+    "-sdk",
+    "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.sdk",
+    "-parse-as-library",
+    "-Xfrontend",
+    "-serialize-diagnostics-path",
+    "-Xfrontend",
+    "/Users/marium.hassan/JustTweak/.build/plugins/cache/Generate_Manual.dia",
```
Comment (albertodebortoli) [inline .build/plugins/cache/Generate_Manual-state.json:34]: looking at user-specific lines like this one, are we sure we want to commit the `.build` folder at all?

File: Framework/Sources/UI/Cells/BooleanTweakTableViewCell.swift [hunk 2]
```diff
@@ -7,9 +7,8 @@ import UIKit
 
 internal class BooleanTweakTableViewCell: UITableViewCell, TweakViewControllerCell {
 
-    var feature: String?
-    var variable: String?
-
+    private var _feature: String? = nil
```
Comment (albertodebortoli) [inline Framework/Sources/UI/Cells/BooleanTweakTableViewCell.swift:10]: it seems that here and in TextTweakTableViewCell we are regressing the changes of #81 
maybe a wrong `stash apply/pop`?

File: Framework/Sources/UI/TweakViewController.swift [hunk 3]
```diff
@@ -231,14 +231,16 @@ extension TweakViewController {
...
     internal func tweakConfigurationCellDidChangeValue(_ cell: TweakViewControllerCell) {
-        guard let featureValue = cell.feature, let variableValue = cell.variable ,
-              let indexPath = indexPathForTweak(with: featureValue, variable: variableValue),
-              let tweak = tweakAt(indexPath: indexPath)
-        else { return }
-        let feature = tweak.feature
-        let variable = tweak.variable
-        tweakManager.set(cell.value, feature: feature, variable: variable)
-        tweak.value = cell.value
+        if let feature = cell.feature, let variable = cell.variable {
```
Comment (albertodebortoli) [inline Framework/Sources/UI/TweakViewController.swift:234]: same here ([change in #80](https://github.com/justeat/JustTweak/pull/81/files#diff-e977a36c2345be54b8b1ada8ab760c787dcf0bc13dc9317669bd3c436512eb7cR234))

File: Tests/Sources/Core/TweakManager+PresentationTests.swift [hunk 4]
```diff
@@ -13,13 +13,13 @@ class TweakManager_PresentationTests: XCTestCase {
     
     var tweakManager: TweakManager!
     let localTweakProviderLowPriority: LocalTweakProvider = {
-        let bundle = Bundle(for: TweakManagerTests.self)
-        let jsonConfigurationURL = bundle.url(forResource: "LocalTweaks_test", withExtension: "json")!
+       // let bundle = Bundle(for: TweakManagerTests.self)
```
Comment (albertodebortoli) [inline Tests/Sources/Core/TweakManager+PresentationTests.swift:16]: can be removed

File: Tests/Sources/Core/TweakManagerTests.swift [hunk 5]
```diff
@@ -6,12 +6,13 @@
...
 @testable import JustTweak
 
+
 class TweakManagerTests: XCTestCase {
     
     var tweakManager: TweakManager!
     let localTweakProvider: TweakProvider = {
-        let bundle = Bundle(for: TweakManagerTests.self)
-        let jsonConfigurationURL = bundle.url(forResource: "LocalTweaks_test", withExtension: "json")!
+       // let bundle = Bundle(for: TweakManagerTests.self)
```
Comment (albertodebortoli) [inline Tests/Sources/Core/TweakManagerTests.swift:14]: can be removed

File: Tests/Sources/Core/TweakManagerTests.swift [hunk 6]
```diff
@@ -88,23 +89,7 @@ class TweakManagerTests: XCTestCase {
         NotificationCenter.default.post(name: TweakProviderDidChangeNotification, object: self, userInfo: userInfo)
         XCTAssertFalse(didCallClosure)
     }
-    
-    func testTweakManagerDecryption() throws {
```
Comment (albertodebortoli) [inline Tests/Sources/Core/TweakManagerTests.swift:92]: are we sure we want to remove this test?

File: Tests/Sources/Core/UserDefaultsTweakProviderTests.swift [hunk 7]
```diff
@@ -6,6 +6,7 @@
 import XCTest
 import JustTweak
 
+
```
Comment (albertodebortoli) [inline Tests/Sources/Core/UserDefaultsTweakProviderTests.swift:9]: probably an accidental change

File: Tests/Sources/UI/TweakViewControllerTests.swift [hunk 8]
```diff
@@ -14,8 +14,8 @@ class TweakViewControllerTests: XCTestCase {
     
     override func setUp() {
         super.setUp()
-        let bundle = Bundle(for: TweakViewControllerTests.self)
-        let jsonURL = bundle.url(forResource: "LocalTweaks_test", withExtension: "json")
+       // let bundle = Bundle(for: TweakViewControllerTests.self)
```
Comment (albertodebortoli) [inline Tests/Sources/UI/TweakViewControllerTests.swift:17]: can be removed

File: .build/plugins/cache/Generate_Manual-state.json (hunk 1 above)
Comment (victor-sarda) [reply .build/plugins/cache/Generate_Manual-state.json:34]: +1, this folder should be git ignored

File: Tests/Sources/Resources/LocalTweaks_test_override.json [hunk 9]
```diff
@@ -1,10 +1,5 @@
 {
   "ui_customization": {
-    "display_blue_view": {
-      "Title": "Display Blue View",
-      "Group": "UI Customization",
-      "Value": true
-    },
```
Comment (victor-sarda) [inline Tests/Sources/Resources/LocalTweaks_test_override.json:7]: Not sure why this one is gone?

**Review by victor-sarda — CHANGES_REQUESTED** [review]
Nice one, just a quick question 👀 
Also, could you please update the readme file? This section is not valid anymore with your changes:
image

File: Tests/Sources/Core/TweakManagerTests.swift (hunk 6 above)
Comment (mariumhassan) [reply Tests/Sources/Core/TweakManagerTests.swift:92]: Previously this test was using "LocalTweaks_example.json" file from DempApp resources, updated the test.

File: Tests/Sources/Resources/LocalTweaks_test_override.json (hunk 9 above)
Comment (mariumhassan) [reply Tests/Sources/Resources/LocalTweaks_test_override.json:7]: I added this only to test the changes, committed by mistake in first Commit.

File: .github/workflows/pull-request-workflow.yml [hunk 10]
```diff
@@ -30,7 +30,7 @@ jobs:
     - name: Run tests (JustTweak)
       run: bundle exec fastlane unit_tests_just_tweak device:'iPhone 11'
     - name: Run tests (TweakAccessorGenerator)
-      run: bundle exec fastlane unit_tests_tweak_accessor_generator
```
Comment (victor-sarda) [inline .github/workflows/pull-request-workflow.yml:33]: I think you need to remove the `pod install` step in this file 👀

File: .github/workflows/pull-request-workflow.yml [hunk 11]
```diff
@@ -22,15 +22,10 @@ jobs:
...
-    - name: Run pod install
-      run: |
-        set -eo pipefail
-        export LIB_VERSION=$(git describe --tags `git rev-list --tags --max-count=1`)
-        bundle exec pod install --project-directory=Example
     - name: Run tests (JustTweak)
       run: bundle exec fastlane unit_tests_just_tweak device:'iPhone 11'
     - name: Run tests (TweakAccessorGenerator)
-      run: bundle exec fastlane unit_tests_tweak_accessor_generator
+      run: bundle exec fastlane unit_tests_tweak_accessor_generator device:'iPhone 11'
```
Comment (victor-sarda) [inline .github/workflows/pull-request-workflow.yml:28]: I don't think you want this. This is a CLI tool that runs on mac, not on iPhone 👀

Comment (mariumhassan) [reply .github/workflows/pull-request-workflow.yml:28]: yess going to remove it.

**Review by victor-sarda — APPROVED** [review]
👏

**Comment by albertodebortoli** [comment]
Hi @mariumhassan, this PR got left open for more than 1 year. We are now moving to Toggles and not investing in this component anymore. I'm gonna go ahead and close this as well as deprecating and archiving the repo.

**Pull request closed without merging**